e2e-tests-rust/src/provider/testing.rs

```diff
@@ -93,6 +101,66 @@ pub async fn init_testing_provider(
     })
 }
 
+// Init testing provider which sends specified HTTP headers e.g. for authentication
+// Outside of `TestingProvider` to avoid specifying `P`
```

Comment:
Can we solve it by using `Box<dyn FullZksyncProvider<T>>` instead of template parameters?
Transport can also be erased by using `Box<dyn Transport + Clone>`.